supercollider/server.py

```diff
@@ -13,188 +16,131 @@ def __init__(self, hostname="127.0.0.1", port=57110):
+
+        self.osc_server_thread = Thread(target=self._osc_server_listen, daemon=True)
         self.osc_server_thread.start()
-        self.handlers = {
-            "/n_set": {},
-            "/b_info": {},
-            "/b_setn" : {},
-            "/done": {},
-            "/status.reply": None,
-            "/version.reply": None,
-            "/synced": None,
-            "/g_queryTree.reply": None
-        }
-
-        #-----------------------------------------------------------------------
-        # Necessary when querying a node ID for add actions.
-        #-----------------------------------------------------------------------
+
+        # Callbacks witout await_response timeout
+        # self.dispatcher.map("/example", self.dummy_handler)
+
+        # For callback timeouts
+        self.event = Event()
+        
+        # SC node ID for Add actions
         self.id = 0
-        try:
-            self.sync()
-        except SuperColliderConnectionError as e:
-            self.osc_server.free()
-            self.osc_server = None
-            raise e
-
-    def __del__(self):
-        if  self.osc_server:
-            self.osc_server.free()
-            self.osc_server = None
-        self.osc_server_thread.join()
-
-    def sync(self):
-        """
-        Causes the server to immediately execute and flush all asynchronous commands.
-        This command is always synchronous so blocks the current thread.
 
-        This can be used as a `ping` functionality to check that the server is running
-        and responding.
-        """
-        self._send_msg("/sync")
-        return self._await_response("/synced", None, lambda n: n)
-    
-    def query_tree(self, group=None):
-        self._send_msg("/g_queryTree", group.id if group else 0, 0)
-        return self._await_response("/g_queryTree.reply")
+        self.sync()
 
-    def get_status(self, callback=None, blocking=True):
-        """
-        Query the current Server status, including the number of active units, CPU
-        load, etc.
-
-        Example:
-            >>> server.status
-            {
-                'num_ugens': 5,
-                'num_synths': 1,
-                'num_groups': 2,
-                'num_synthdefs': 107,
-                'cpu_average': 0.08170516043901443,
-                'cpu_peak': 0.34912213683128357,
-                'sample_rate_nominal': 44100.0,
-                'sample_rate_actual': 44100.07866992249
-            }
-        """
-        def _handler(args):
-            args_dict = {
+    # ------------------- Handlers ------------------- #
+    # TODO: Move handlers to a separate file or class
+    def dummy_handler(self, address, *args):
+        pass
+
+    def simple_handler(self, address, *args):
+        return args
+
+    def param_handler(self, address, *args):
+        return args[2]
+
+    def buf_handler(self, address, *args):
+        return args[3:]
+
+    def status_handler(self, address, *args):
```

Comment:
I'm not keen on having instance methods for these "handler" functions, for a few reasons:
 - it clutters up the `Server` namespace with methods that are only designed to be used internally (try `pydoc supercollider.Server` and you will see all of these methods showing up)
 - it distributes the message-parsing logic over different areas of the code; e.g. in the case of `get_version`, all of the logic around dict construction was previously encapsulated in the `get_version` function
 - in the case of `buf_handler`, it means that other classes (e.g. `Buffer`) also have to utilise message parsing functions from Server

I'd prefer to return to the previous approach of implementing local `_handler` functions within the methods that utilise them. Thanks!

Reply:
Done. I have made all handlers internal again in [0be7cd2](https://github.com/ideoforms/python-supercollider/pull/12/commits/0be7cd2a75c6e378e7968ac64c7aca7b7bfbf96b)

I've also noticed that the cases where blocking=False were still expecting `pyliblo` handlers. [910e857](https://github.com/ideoforms/python-supercollider/pull/12/commits/910e8578e8da3afefdbf9b12a30d440eedc5afab)